SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -604,33 +604,41 @@ void FSpatialNetGUIDCache::RemoveEntityNetGUID(Worker_EntityId EntityId)
 		}
 	}
 
-	// Remove dynamically attached subobjects
+	// Remove dynamic subobjects and objects replicated on the actorchannel.
 	if (USpatialActorChannel* Channel = SpatialNetDriver->GetActorChannelByEntityId(EntityId))
 	{
-		for (UObject* DynamicSubobject : Channel->CreateSubObjects)
+		for (auto& ObjReplicatorPair : Channel->ReplicationMap)
 		{
-			if (FNetworkGUID* SubobjectNetGUID = NetGUIDLookup.Find(DynamicSubobject))
+			TWeakObjectPtr<UObject> WeakObjectPtr = ObjReplicatorPair.Value->GetWeakObjectPtr();
+			if (WeakObjectPtr.IsValid())
 			{
-				if (FUnrealObjectRef* SubobjectRef = NetGUIDToUnrealObjectRef.Find(*SubobjectNetGUID))
+				if (FNetworkGUID* ObjectNetGUID = NetGUIDLookup.Find(WeakObjectPtr))
 				{
-					UnrealObjectRefToNetGUID.Remove(*SubobjectRef);
-					NetGUIDToUnrealObjectRef.Remove(*SubobjectNetGUID);
+					if (FUnrealObjectRef* SubobjectRef = NetGUIDToUnrealObjectRef.Find(*ObjectNetGUID))
+					{
+						UnrealObjectRefToNetGUID.Remove(*SubobjectRef);
+						NetGUIDToUnrealObjectRef.Remove(*ObjectNetGUID);
+					}
 				}
 			}
 		}
 	}
 
-	// Remove actor.
+	// Remove actor mappings.
 	FNetworkGUID EntityNetGUID = GetNetGUIDFromEntityId(EntityId);
-	// TODO: Figure out why NetGUIDToUnrealObjectRef might not have this GUID. UNR-989
-	if (FUnrealObjectRef* ActorRef = NetGUIDToUnrealObjectRef.Find(EntityNetGUID))
-	{
-		UnrealObjectRefToNetGUID.Remove(*ActorRef);
-	}
-	NetGUIDToUnrealObjectRef.Remove(EntityNetGUID);
-	if (StablyNamedRefOption.IsSet())
-	{
-		UnrealObjectRefToNetGUID.Remove(StablyNamedRefOption.GetValue());
+	if (EntityNetGUID.IsValid()) {
+		// This may have already been removed by the code removing items replicated on the ActorChannel.
+
+		// TODO: Figure out why NetGUIDToUnrealObjectRef might not have this GUID. UNR-989
+    	if (FUnrealObjectRef* ActorRef = NetGUIDToUnrealObjectRef.Find(EntityNetGUID))
+    	{
+    		UnrealObjectRefToNetGUID.Remove(*ActorRef);
+    	}
+    	NetGUIDToUnrealObjectRef.Remove(EntityNetGUID);
+    	if (StablyNamedRefOption.IsSet())
+    	{
+    		UnrealObjectRefToNetGUID.Remove(StablyNamedRefOption.GetValue());
+    	}
```

Comment:
tab

Reply:
ye whoops, forgot to push the formatting commit